openedx/core/djangoapps/cookie_metadata/middleware.py

```diff
@@ -0,0 +1,59 @@
+"""Middleware to changes name of an incoming cookie"""
+from django.conf import settings
+
+
+class CookieNameChange:
+    """Changes name of an incoming cookie"""
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        """
+        Changes the names of a cookie in request.COOKIES
+
+        For this middleware to run:
+        - set COOKIE_NAME_CHANGE_ACTIVATE_EXPAND_PHASE =  True
+        - COOKIE_NAME_CHANGE_EXPAND_INFO is a dict and has following info:
```

Comment:
I'm tempted to say this should be a list of dicts, but I guess we don't have any plans to migrate more than one cookie at a time, and these are supposed to be short-lived... Still, it would be a pretty powerful tool to use in an emergency. Up to you.

Reply:
We decided to keep the scope of this middleware small and remove it after we are done to not add more clutter to edx-platform. While, yes, it would be a powerful tool, considering it is likely to be rarely used, it is not worth keeping around.

Reply:
Makes sense!